src/foremast/pipeline/construct_pipeline_block.py

```diff
@@ -100,21 +126,62 @@ def construct_pipeline_block(env='',
         dict: Pipeline JSON template rendered with configurations.
 
     """
-    LOG.info('%s block for [%s].', env, region)
-
-    if env.startswith('prod'):
-        template_name = 'pipeline/pipeline_{}.json.j2'.format(env)
-    else:
-        template_name = 'pipeline/pipeline_stages.json.j2'
 
+    LOG.info('%s block for [%s].', env, region)
     LOG.debug('%s info:\n%s', env, pformat(settings))
 
     gen_app_name = generated.app_name()
-    user_data = generate_encoded_user_data(env=env, region=region, app_name=gen_app_name, group_name=generated.project)
+    data = copy.deepcopy(settings)
+
+    setup_kwargs = {
+        'appname': gen_app_name,
+        'settings': settings,
+        'env': env,
+        'region': region,
+        'project': generated.project
+    }
+
+    if pipeline_type == 'ec2':
+        setup_kwargs['region_subnets'] = kwargs['region_subnets']
+        data = ec2_pipeline_setup(**setup_kwargs)
```

Comment:
If we're not reusing or modifying the keyword arguments, we should pass in keyword arguments instead of unpacking the `dict`.